packages/amplify-category-api/provider-utils/awscloudformation/service-walkthroughs/apigw-walkthrough.js

```diff
@@ -492,53 +492,37 @@ async function askPaths(context, answers, currentPath) {
 }
 
 function validatePathName(name, paths) {
-  const err = null;
-
-  if (name.length === 0 || name.substring(name.length - 1) === '/') {
-    return 'Each sub-path must begin with a letter or number.';
+  if (name.length === 0) {
+    return 'The path must not be empty';
   }
 
-  // Set / as a first character of path name
-  if (name.substring(0, 1) !== '/') {
-    return 'Path must begin with / e.g. /items';
-  }
-  if (/[^a-zA-Z0-9\-/]/.test(name)) {
-    return 'You can use the following characters: a-z A-Z 0-9 - /';
+  if (name.charAt(name.length - 1) === '/') {
+    return 'The path must not end with /';
   }
 
-  // If the are is something like /asasd//asa must be detected
-  // Splitting the string with / to find empty sub-path
-  const split = name.split('/');
-  for (let i = 1; i < split.length; i += 1) {
-    const val = split[i];
-    if (val.length === 0) {
-      return 'Each sub-path must begin with a letter or number';
-    }
+  if (name.charAt(0) !== '/') {
+    return 'The path must begin with / e.g. /items';
   }
 
-  // Checking if there is already that path created on the API
-  if (paths.find(path => path.name === name)) {
-    return 'Path name already exists';
+  // Matches parameterized paths such as /book/{isbn}/page/{pageNum}
+  // This regex also catches the above conditions, but those are left in to provide clearer error messages.
+  if (!/^(?:\/(?:[a-zA-Z0-9\-]+|{[a-zA-Z0-9\-]+}))+$/.test(name)) {
+    return 'Each path part must use characters a-z A-Z 0-9 - and must not be empty.\nOptionally, a path part can be surrounded by { } to denote a path parameter.';
   }
 
-  // Create subpath from the beginning to find a match on existing paths
+  const split = name.split('/').filter(sub => sub !== ''); // because name starts with a /, this filters out the first empty element
+  // Check if any prefix of this path matches an existing path
   const findSubPath = (path, subpath) => path.name === subpath;
   let subpath = '';
-  split.forEach(sub => {
+  const subMatch = split.some(sub => {
     subpath = `${subpath}/${sub}`;
-    const subpathFind = paths.find(path => findSubPath(path, subpath));
-    if (subpathFind) {
-      return `A different path already matches this sub-path: ${subpath}`;
-    }
+    return paths.map(path => path.name).find(name => name === subpath) !== undefined;
   });
-
-  // Check if other paths are a subpath of the new path
-  subpath = paths.find(path => path.name.indexOf(name) === 0);
-  if (subpath) {
-    return `An existing path already match with the one provided: ${subpath.name}`;
+  if (subMatch) {
+    return `An existing path already matches this sub-path: ${subpath}`;
   }
 
-  return err;
+  return null;
```

Comment:
Is there a reason we are returning null instead?

Reply:
At the top of the function err was set to null and never reassigned. It seems clearer to just return null directly